src/main.c

```diff
@@ -3293,6 +3376,13 @@ H2O_NORETURN static void *run_loop(void *_thread_index)
             conf.listeners[i]->quic.thread_fds[thread_index] = fds[1];
         }
     }
+
+    /* setup the h2olog listener */
+    if (conf.h2olog.listen_fd != -1) {
+        conf.h2olog.listen_sock =
+            h2o_evloop_socket_create(conf.threads[thread_index].ctx.loop, conf.h2olog.listen_fd, H2O_SOCKET_FLAG_DONT_READ);
```

Comment:
```
        h2o_socket_t *listener =
            h2o_evloop_socket_create(conf.threads[thread_index].ctx.loop, conf.h2olog.listen_fd, H2O_SOCKET_FLAG_DONT_READ);
        h2o_socket_read_start(listener, ...);
```
As we want the listener to always be active, we can call `h2o_socket_read_start` here and nowhere else. That means that we do not need to have a global variable of type `h2o_socket_t *`(i.e., `listen_sock`).